### ISSUE TITLE
Faasd issues on Debian Bullseye 

### ISSUE THREAD

**Issue by waalge** [openfaas/faasd#235]
## Expected Behaviour

Trying to adapt [blog][alexellis_post] [posts][myedes_post] to run faasd. 

**Expected:** Faasd to be working. 

## Current Behaviour

Had [iptables](https://github.com/openfaas/faasd/issues/227#issuecomment-1003526057) and found the same resolution as noted there. 

Had errors from journal reported as 
```
Jan 17 22:14:01 raspberrypi faasd[9803]: 2022/01/17 22:14:01 Error creating task: failed to create shim: OCI runtime create failed: container_linux.go:367: starting container process caused: process_linux.go:495: container init caused: rootfs_linux.go:76: mounting "/var/lib/faasd/nats" to rootfs at "/nats" caused: stat /var/lib/faasd/nats: no such file or directory: unknown
Jan 17 22:14:01 raspberrypi faasd[9803]: Error: failed to create shim: OCI runtime create failed: container_linux.go:367: starting container process caused: process_linux.go:495: container init caused: rootfs_linux.go:76: mounting "/var/lib/faasd/nats" to rootfs at "/nats" caused: stat /var/lib/faasd/nats: no such file or directory: unknown
Jan 17 22:14:01 raspberrypi systemd[1]: faasd.service: Main process exited, code=exited, status=1/FAILURE
```
I think this occurred initially for all services, but then perhaps after rebooting  `basic-auth-plugin` claimed to be successful
while NATS and prometheus errored as above 

I stopped the `faasd.service` and tried  running the command `faasd up` from `/var/lib/faasd` (from within `su` context) 
This then arrived at a new error
```
unable to dial to 10.62.0.10:8080, error: dial tcp 10.62.0.10:8080: connect: no route to host
```
This error was also reached when I started the service again. 

According to the roadmap, this is a known issue, but numerous reboots, restarts etc. as suggested didn't resolve the problem. 

Somewhere a long the lines, I edited the boot files to include `cgroup_enable=cpuset cgroup_memory=1 cgroup_enable=memory` as required by other container-y management things. 

 

[alexellis_post]: https://blog.alexellis.io/faasd-for-lightweight-serverless/
[myedes_post]: https://myedes.io/serverless-on-raspberrypi/

## Are you a GitHub Sponsor (Yes/No?)

- [x] No

## List all Possible Solutions

Rollback to the versions in the walk through but... Not really the spirit. 

## Steps to Reproduce (for bugs)

I try to keep all the relevant commands in a shell script,
running a section at a time, and the commenting it out. 
This is from a a pretty fresh install of [this os][rpi_os] purging X. 
Obvs this doesn't include some of the prodding described above.

[rpi_os]: https://downloads.raspberrypi.org/raspios_arm64/images/raspios_arm64-2021-11-08/

```
## SYSTEM VARS 

export CONTAINERD_VERSION=1.5.7
export CONTAINER_NETWORKING_VERSION=1.0.1
export ARCH=arm64

## GET CONTAINERD BINARIES 

# export ZIP_FILE=containerd-${CONTAINERD_VERSION}-linux-${ARCH}.tar.gz
# curl -sSL https://github.com/alexellis/containerd-arm/releases/download/v${CONTAINERD_VERSION}/${ZIP_FILE} | sudo tar -xvz -C /usr/local/bin/
 
## SETUP CONTAINERD SERVICE

# curl -SLfs https://raw.githubusercontent.com/containerd/containerd/v${CONTAINERD_VERSION}/containerd.service | sudo tee /etc/systemd/system/containerd.service

## INSATLL FAAS-CLI

# curl -sLfS https://cli.openfaas.com | sudo sh

# source <(faas-cli completion --shell bash)

## SETUP NETWORKING 

# sudo modprobe br_netfilter && \
# sudo sysctl net.bridge.bridge-nf-call-iptables=1
# 
# # Make it permanent
# echo "br_netfilter" | sudo tee -a /etc/modules-load.d/modules.conf
# 
# echo "net.bridge.bridge-nf-call-iptables=1" | sudo tee -a /etc/sysctl.conf

## INSTALL CONTAINER NETWORKING 

# sudo mkdir -p /opt/cni/bin
# 
# curl -sSL https://github.com/containernetworking/plugins/releases/download/v${CONTAINER_NETWORKING_VERSION}/cni-plugins-linux-${ARCH}-v${CONTAINER_NETWORKING_VERSION}.tgz | sudo tar -xz -C /opt/cni/bin
# 
# ## MORE NETWORKING 
# 
# sudo /sbin/sysctl -w net.ipv4.conf.all.forwarding=1
# 
# echo "net.ipv4.conf.all.forwarding=1" | sudo tee -a /etc/sysctl.conf

## GET FAASD 

# export FAASD_VERSION=0.14.4
# 
# sudo curl -sSLf "https://github.com/openfaas/faasd/releases/download/${FAASD_VERSION}/faasd-${ARCH}" \
#     --output "/usr/local/bin/faasd" \
#     && sudo chmod a+x "/usr/local/bin/faasd"

export GOPATH=$HOME/go/
mkdir -p $GOPATH/src/github.com/openfaas
cd $GOPATH/src/github.com/openfaas

git clone https://github.com/openfaas/faasd --depth=1
cd faasd
sudo faasd install
cd ~/
```

## Context

I just wanna checkout what minimal faas looks like 

I've previously run a k3 cluster on Pis. I play with devops, but am on the noob end of the spectrum.

## Your Environment

* OS and architecture:

* Versions:

```sh
$ sh -v check-versions.sh 
go version
check-versions.sh: 1: go: not found

containerd -version
containerd github.com/containerd/containerd v1.5.7 8686ededfc90076914c5238eb96c883ea093a8ba

uname -a
Linux raspberrypi 5.10.63-v8+ #1488 SMP PREEMPT Thu Nov 18 16:16:16 GMT 2021 aarch64 GNU/Linux

cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux 11 (bullseye)"
NAME="Debian GNU/Linux"
VERSION_ID="11"
VERSION="11 (bullseye)"
VERSION_CODENAME=bullseye
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"

faasd version
  __                     _ 
 / _| __ _  __ _ ___  __| |
| |_ / _` |/ _` / __|/ _` |
|  _| (_| | (_| \__ \ (_| |
|_|  \__,_|\__,_|___/\__,_|

faasd version: 0.14.4	commit: 8fbdd1a461196520de75fe35ac0b5bdda6403ac7

```

**Comment by alexellis** [comment]
Hi, I would suggest using Ubuntu 20.04 LTS or the Buster OS instead of Debian Bullseye. There's no advantage that I'm aware of that you would find by using Bullseye instead.

If you're using a Raspberry Pi, there are known issues with the OS as it's being bundled at present.

Closing at a duplicate of: #227

Alex

**Comment by waalge** [comment]
@alexellis Thanks.
Although I don't this is not a duplicate.

**Comment by alexellis** [comment]
I'm happy for you to explain why you think that both issues are not about compatibility with Debian Bullseye?

**Comment by waalge** [comment]
@alexellis It is a "faasd may have a compatibility issue with Debian Bullseye" #227 
but is not about iptables which is resolved with installing iptables. 
#227 doesn't cover subsequent issues. 

Although 
```
unable to dial to 10.62.0.10:8080, error: dial tcp 10.62.0.10:8080: connect: no route to host
```
Is known, the suggested solution did not work for me - the error consistently appears. 

Very happy to do any further tests if its helpful. 
Otherwise I'll switch distro.

**Comment by alexellis** [comment]
The community member on #227 explained that it worked for him after installing iptables, so I would encourage you to ask @rvramesh what you may be doing differently?

> I'm happy for you to explain why you think that both issues are not about compatibility with Debian Bullseye?

Our recommendation is for Buster on RPi or Ubuntu, why do you feel Bullseye is better, or necessary?

I'm sure we will get to this at some point in the future, if the community wants to spend the time on it.

**Comment by rvramesh** [comment]
> git clone https://github.com/openfaas/faasd --depth=1
> cd faasd
> sudo faasd install

@waalge I got it working by using `sudo ./hack/install.sh` instead of `sudo faasd install` . Have you tried doing this?

**Comment by waalge** [comment]
> why do you feel Bullseye is better, or necessary?

I didn't apriori foresee this being an issue when I started. 
Adapting walkthroughs to accommodate newer versions is often an achievable, and in case, informative process.

I searched as best I could the github issues for errors akin to mine, and I didn't find any (installing iptables wasn't really an obstacle). 
I don't mind this being closed - without pointers I can't assist in a fix. 
But I don't think this is a duplicate, otherwise I wouldn't have submitted an issue. 

---

@rvramesh That I have not. 
I will give that a go... :)
